### PR TITLE
Added edge handling in LUT interp with pre-built LUT

### PR DESCRIPTION
**Pull request by evan-greenbrg**
Isofit errors out when using a pre-built lut grid, and the interpolated value (as noted in the "interp" key) is at, or smaller than the smallest value in the lut dimension for that key. See example for "surface_elevation_km" below:

```
                 "lut_names": {
                        "AERFRAC_2": {
                            "gte": 0.001,
                            "lte": 1.0
                        },
                        "H2OSTR": {
                            "gte": 0.05,
                            "lte": 5.0
                        },
                        "observer_zenith": {
                            "gte": 2.2024,
                            "lte": 18.9262
                        },
                        "relative_azimuth": {
                            "gte": 0.0002,
                            "lte": 86.0982
                        },
                        "surface_elevation_km": {
                            "interp": 0.0
                        }
                    },
```

As noted in #697, we may not want to support run cases when the scene-value is lower than the LUT value, but at the bare minimum we have to support cases when the scene-value is _at_ the LUT grid minimum, which currently fails.

The root of the issue is with how we slice the LUT dimension. For example, when the interpolated value (`val`) is equal to the smallest value in the lut grid (`dim[0]`):
``` 
b = np.searchsorted(dim * orientation, val * orientation)
```
returns 0 and
```
slice(b - 1, b + 1)
```
returns a slice, which always indexes to an empty list. 

Interestingly, the inverse is not true. If the interpolated value is greater than the largest value in the lut grid (`val > dim[-1]`):
``` 
b = np.searchsorted(dim * orientation, val * orientation)
```
returns len(dim) and
```
slice(b - 1, b + 1)
```
returns a slice for the last LUT dim value. It looks like we actually currently allow for scene-values for LUT elements to be greater than the maximum value within the LUT grid.

The proposed changes are to:

1. Handle the specific cases directly when the interpolated values are at the minimum and maximum LUT grid dimension values.
2. To be discussed, but we could raise exceptions when the interpolation is trying to resolve values outside of the LUT grid dimension values. Not sure how necessary this is given that it should be an edge case, and in test cases when using pre-built LUTs it may be advantageous to run outside the bounds.

### CODE REVIEW

File: isofit/core/isofit.py [hunk 1]
```diff
@@ -249,6 +265,15 @@ def run_set_of_spectra(self, indices: np.array):
                         fill_value="extrapolate",
                     )
                     input_data.meas = interp(self.io.meas_wl)
+
```
Comment (evan-greenbrg) [inline isofit/core/isofit.py:268]: This is somewhat outside of the scope of the PR, but I wanted to raise this as a potential point of confusion, which I found while trying to debug this issue.

The extractions script adds a label == 0, which then results in the subs_rdn file having an entry at 0, 0 where all values are 0. 
```
    lbl_img = envi.open(envi_header(lbl_file), lbl_file)
    labels = lbl_img.read_band(0)
    un_labels = np.unique(labels).tolist()
    if 0 not in un_labels:
        un_labels.insert(0, 0)
    nout = len(un_labels)
```
We don't explicitly handle this row within Isofit.run. Why do we insert a 0 index into the extracted subs-level files? Is it to handle ORT images? 

I wanted to call out this code block, which handles all-zero rows within Isofit.run, and rather than change the extractions, skip empty lines.

Comment (pgbrodrick) [reply isofit/core/isofit.py:268]: You're right that it's confusing, and handled a little subtly.  It's there to handle the ortho data, as you suspected.   There are probably (almost certainly) other ways to handle this...though I seem to recall when I went to switch one out a number of years ago it was trickier than I thought.
